lib/controller.dart

```diff
@@ -25,6 +25,21 @@ class Controller {
   CustomThemes defaultTheme = CustomThemes.Dark;
   Locale defaultLocale = AppLocalizations.supportedLocales.first;
 
+  late RuleService _ruleService;
+  late MissionService _missionService;
+
+  List<Team> teams = [];
+  final List<Mission> missions = [];
+  final List<RuleChapter> rules = [];
+
+  Controller({
+    required RuleService ruleService,
```

Comment:
required this._ruleService ? ou ca marche pas avec les attributs privés ?

réel besoin de le passer en paramètre ? ou ca pourrait être instancié par le constructeur ?

Reply:
Non, ce n'est pas possible : `Named parameters can't start with an underscore.`

Je les passe en paramètre pour pouvoir les surcharger par les mocks dans les tests unitaires au final.

Reply:
https://pub.dev/packages/get_it ? Mdr

Reply:
`AeContainer.Resolve` ? 😄

Reply:
Ca peut potentiellement être plus simple, si en plus j'ai juste à mocker le `GetIt`. Je tente dans cette PR ou dans une autre ?

Reply:
ok pour une autre. histoire de séparer TU de DI